### PR TITLE
helm: add cluster proxy rbac for admin when deploy Karmada control plane

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
helm: add cluster proxy rbac for admin when deploy Karmada control plane
**Which issue(s) this PR fixes**:
Part of #1546 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: charts/templates/cluster-proxy-admin-rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  name: {{ $name }}-cluster-proxy-admin
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ $name }}-cluster-proxy-admin
+subjects:
+  - kind: User
+    name: "system:admin"
+
+{{- end }}
```
Comment (XiShanYongYe-Chang) [inline charts/templates/cluster-proxy-admin-rbac.yaml:29]: Add a blank line below the line.

Comment (AllenZMC) [reply charts/templates/cluster-proxy-admin-rbac.yaml:29]: @XiShanYongYe-Chang  Why need a blank line below the line?

Comment (lonelyCZ) [reply charts/templates/cluster-proxy-admin-rbac.yaml:29]: I guess it is a encoding specification.

Comment (AllenZMC) [reply charts/templates/cluster-proxy-admin-rbac.yaml]: @XiShanYongYe-Chang  I have modified it.

**Comment by XiShanYongYe-Chang** [comment]
I'm not familiar with helm, ask @jrkeen @pidb to help review.

File: charts/templates/cluster-proxy-admin-rbac.yaml (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply charts/templates/cluster-proxy-admin-rbac.yaml]: > I guess it is a encoding specification.

It's like this.

**Comment by pidb** [comment]
> I'm not familiar with helm, ask @jrkeen @pidb to help review.

Okay, Thank you for your pr.

**Comment by pidb** [comment]
> /assign @jrkeen @pidb

It looks great to me, but may you provide a test?

**Comment by AllenZMC** [comment]
> 

How to test, please? I'm testing on this machine and can generate the correct yaml file.

**Comment by lonelyCZ** [comment]
I am also not familiar with helm, but BTW  `proxy-rbac` could be deployed in `karmada-apiserver` rather than host cluster.

**Comment by AllenZMC** [comment]
`
 helm template test --debug ./
---
# Source: karmada/templates/cluster-proxy-admin-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: test-cluster-proxy-admin
rules:
  - apiGroups:
      - 'cluster.karmada.io'
    resources:
      - clusters/proxy
    verbs:
      - '*'
---
# Source: karmada/templates/cluster-proxy-admin-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: test-cluster-proxy-admin
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: test-cluster-proxy-admin
subjects:
  - kind: User
    name: "system:admin"
`

**Comment by AllenZMC** [comment]
cc @RainbowMango

**Review by pidb — APPROVED** [review]
I look ok

**Review by RainbowMango — COMMENTED** [review]
/lgtm

**Comment by fanzhihai0215** [comment]
> I am also not familiar with helm, but BTW `proxy-rbac` could be deployed in `karmada-apiserver` rather than host cluster.

Currently, on master branch, [cluster-proxy-admin-rbac.yaml](https://github.com/karmada-io/karmada/blob/master/charts/templates/cluster-proxy-admin-rbac.yaml) is located in charts/templates. I think it will be deployed in `host cluster` rather than `karmada-apiserver`.  

Maybe, i think `proxy-rbac` should be deployed as `static-resources` in `post-install-job.yaml`, like `webhook-configuration.yaml` and `system-namespace.yaml`. 

Can anyone help with confirmation?

@lonelyCZ @XiShanYongYe-Chang

**Comment by lonelyCZ** [comment]
Good finding! Could you help to test installation of karmada by helm? @fanzhihai0215 

And `proxy-rbac` should be deployed certainly in `karmada-apiserver`, if there is a problem, welcome to file a pr to fix it.

**Comment by XiShanYongYe-Chang** [comment]
> Good finding! Could you help to test installation of karmada by helm? @fanzhihai0215
> 
> And `proxy-rbac` should be deployed certainly in `karmada-apiserver`, if there is a problem, welcome to file a pr to fix it.

+1

@fanzhihai0215 thanks!

**Comment by fanzhihai0215** [comment]
> Good finding! Could you help to test installation of karmada by helm? @fanzhihai0215
> 
> And `proxy-rbac` should be deployed certainly in `karmada-apiserver`, if there is a problem, welcome to file a pr to fix it.

If it's not urgent, I can try. It may take a few days. I don't have a ready environment, I may test on kind. :)

**Comment by XiShanYongYe-Chang** [comment]
Hi @fanzhihai0215, we can open an issue to facilitate tracing. Can you help make it?

**Comment by AllenZMC** [comment]
I checked it and lost the process of judging the installation mode.

**Comment by fanzhihai0215** [comment]
> Hi @fanzhihai0215, we can open an issue to facilitate tracing. Can you help make it?

I just confirmed. `proxy-rbac` is in `host cluster` rather than `karmada-apiserver`. I can open an issue to fix it.

```
# kubectl --kubeconfig karmada-host.config -n karmada-system get clusterrole
NAME                                                                   CREATED AT
admin                                                                  2022-04-08T08:36:12Z
cluster-admin                                                          2022-04-08T08:36:12Z
edit                                                                   2022-04-08T08:36:12Z
karmada-cluster-proxy-admin                                            2022-04-08T09:23:12Z
......
```
cmd `kubectl --kubeconfig karmada-apiserver.config get clusterrole | grep proxy` return nothing.

BTW, only karmada-apiserver has `MutatingWebhookConfiguration` resources.
```
# kubectl --kubeconfig karmada-host.config get MutatingWebhookConfiguration
No resources found
# kubectl --kubeconfig karmada-apiserver.config get MutatingWebhookConfiguration
NAME              WEBHOOKS   AGE
mutating-config   5          23m
```

**Comment by fanzhihai0215** [comment]
> I checked it and lost the process of judging the installation mode. Thanks @fanzhihai0215 I have submitted a pr.

I think you have a little understanding problem. To a certain extent, `proxy-rbac` should be deployed after karmada installed. It should be deployed with post-install-job.yaml.

**Comment by AllenZMC** [comment]
> > I checked it and lost the process of judging the installation mode. Thanks @fanzhihai0215 I have submitted a pr.
> 
> I think you have a little understanding problem. To a certain extent, `proxy-rbac` should be deployed after karmada installed. It should be deployed with post-install-job.yaml.



> > I checked it and lost the process of judging the installation mode. Thanks @fanzhihai0215 I have submitted a pr.
> 
> I think you have a little understanding problem. To a certain extent, `proxy-rbac` should be deployed after karmada installed. It should be deployed with post-install-job.yaml.

You are right.

**Pull request merged**